### PR TITLE
Fix retry policy support

### PR DESCRIPTION
**Pull request by Gallaecio**
Fixes https://github.com/scrapy-plugins/scrapy-zyte-api/issues/43

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,12 +8,14 @@ repos:
     hooks:
     - id: black
       language_version: python3.8
+      additional_dependencies:
+      - click<8.1
```
Comment (Gallaecio) [inline .pre-commit-config.yaml:12]: Fixes https://github.com/psf/black/issues/2964

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -8,12 +8,14 @@ repos:
...
 -   repo: https://gitlab.com/pycqa/flake8
     rev: 3.8.4
     hooks:
         - id: flake8
 -   repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.812
+    rev: v0.971
```
Comment (Gallaecio) [inline .pre-commit-config.yaml:18]: Addresses https://github.com/python/mypy/issues/5887 for `scrapy_zyte_api.responses.ZyteAPIMixin`.

**Review by BurnzZ — APPROVED** [review]
LGTM @Gallaecio 🙌

**Review by kmike — APPROVED** [review]
Looks good, thanks @Gallaecio!

**Pull request merged**